_posts/2024-06-03-geoserver-tutorials.md

```diff
@@ -0,0 +1,94 @@
+- Now switch to the **Users** tab. Obviously, it lists the only existing user, that is, admin, as shown in the screen.
+- Click on the **Add new user** link, and add `user_admin` with a password of your choice, as Data Administrator.
+- Add "user_admin" to the "group_admins", then press the **Save** button.
+- Repeat the previous step to create a user, `user_editor` as a member of the "group_editors" group, and `user_reader` as a "group_readers" group member. Your list now shows the three users.
+
+We just created three users for the three groups and this may seem overkill to you. Consider them as templates for real users. In the real world, we do not want to have too many administrators; we will probably need several "user_readers" and "user_editors" processing the data. Now, we need to define what they can do on GeoServer.
+
+## Defining roles
+A user or a group without any role assigned is useless. Now it is time to create roles and assign them to our users. Please refer to the following points:
+- From the **User, Groups, and Roles** section, select the **Roles** tab. You will find that two roles already exist. They are the administrative roles assigned to the admin account, and they grant access to all GeoServer configurations. Click on the **Edit** link
+- Switch to the **Roles** tab, then click on **Add new role**.
+- Enter `role_reader` as a new role name. We do not need a Parent role. A child role inherits all the grants from the Parent role, making it useful when you want to extend a basic role with more grants. Indeed, we will do this in the next step. 
+- Press the **Save** button and then repeat the previous step to create the `role_editor` role. This time, select "role_reader" as the Parent role.
+- Press the **Save** button and then repeat the previous step to create the `role_admin` role. This time, select "role_editor" as the Parent role.
+- The final step is to associate a role to users or groups. Select the **User, Groups, and Roles** page from the left pane, then select the **Groups** list and click on the "group_readers" group to edit it. Add the "role_reader" role to the group and save it.
+- Now click on the "group_editors" group and associate it with the role_editor role.
+- Finally, associate the "group_admins" group to the "role_admin" role.
+
+By defining roles and associating them to the users, we completed the definition of our organization. Now, we need to explore how data is bound to roles and users.
+
+## Accessing data and services
+GeoServer supports access and control, both at the service level, allowing for the lockdown of service operations to only authenticated users who have been granted a particular role, and on a per-layer basis. The two approaches cannot be mixed. If you lock down a service to a role, you cannot grant the access on a specific layer to the same role. When working with layers, you can define rules that specify what a role can do on any specific layer.
```

Comment:
` on a per-layer basis`, or per-workspace

`the two approaches cannot be mixed`.  I disagree with this statement.  They are mixed, and they are both considered in series.  Both services rules and data rules must pass.

Reply:
I fixed it.